common/src/main/java/bisq/common/file/FileUtils.java

```diff
@@ -262,13 +286,35 @@ public static void makeFile(File file) throws IOException {
         }
     }
 
+    public static FileOutputStream newFileOutputStreamWithPermissions(String name) throws IOException {
+        Path path = Paths.get(name);
+        try {
+            Set<PosixFilePermission> perms = EnumSet.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE);
+            if (!Files.exists(path)) {
+                Files.createFile(path, PosixFilePermissions.asFileAttribute(perms));
+            } else {
+                Files.setPosixFilePermissions(path, perms);
+            }
+            return new FileOutputStream(path.toFile());
+        } catch (UnsupportedOperationException e) {
```

Comment:
I prefer to check whether the file system supports POSIX before modifying file permissions. Since the Bisq application runs on Windows, relying on exceptions as a fallback could negatively impact performance.